src/main/java/htsjdk/tribble/FeatureReader.java

```diff
@@ -27,16 +27,48 @@
  * @param <T> a feature type
  */
 public interface FeatureReader<T extends Feature> extends Closeable {
-    
-    public CloseableTribbleIterator<T> query(final String chr, final int start, final int end) throws IOException;
 
-    public CloseableTribbleIterator<T> iterator() throws IOException;
+    /**
+     * Query the reader for a particular interval corresponding to a contig and a 1-based closed
+     *
+     * @param chr the contig to be queried
+     * @param start the start of the interval (1-based) to be queried
+     * @param end the last base in the interval to be queried
+     * @return an iterator containing the features that at in the interval.
+     * @throws IOException If there's a problem reading or if the reader is not queryable, e.g. if it doesn't have an index.
+     */
+    CloseableTribbleIterator<T> query(final String chr, final int start, final int end) throws IOException;
 
+    /**
+     * Provides access to all the features in the reader
+     * @return an iterator to all the features in the reader
+     * @throws IOException If there's a problem reading.
+     */
+    CloseableTribbleIterator<T> iterator() throws IOException;
+
+    /**
+     * Closes the reader
+     * @throws IOException
+     */
     @Override
-    public void close() throws IOException;
+    void close() throws IOException;
 
-    public List<String> getSequenceNames();
+    /**
+     * Provides the list of sequenceNames if known. Otherwise will return an empty list.
+     * @return  the list of sequenceNames if known. Otherwise will return an empty list.
+     */
+    List<String> getSequenceNames();
 
-    public Object getHeader();
+    /**
+     * Provide access to the header of the reader
+     * @return the header of the reader
```

Comment:
It can return `null` if there is no header, no? If so, add to the javadoc...

Reply:
ok.